libportability-gfx/src/impls.rs

```diff
@@ -1720,7 +1718,11 @@ pub extern "C" fn gfxCreateGraphicsPipelines(
             })
         };
 
-        let vp_state = unsafe { &*info.pViewportState };
+        let vp_state = if rasterizer_discard {
+            None
+        } else {
+            Some(unsafe { &*info.pViewportState })
+        };
         let empty_dyn_states = [];
         let dyn_states = match unsafe { info.pDynamicState.as_ref() } {
             Some(state) => unsafe {
```

Comment:
could also do `Some(state) if !rasterizer_discard =>` here to avoid checks to follow